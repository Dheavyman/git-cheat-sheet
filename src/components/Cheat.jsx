import React from 'react';

/**
 * Cheat component
 *
 * @param {string} description - Cheat description
 * @param {string} command - Cheat command
 *
 * @returns {object} React element
 */
const Cheat = ({ cheat: { description, command } }) => (
  <div className='cheat'>
    <p className="cheat-description">{description}</p>
    <p className="cheat-command">$&nbsp;{command}</p>
  </div>
);

export default Cheat;
