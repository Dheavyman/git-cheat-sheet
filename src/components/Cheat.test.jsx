import React from 'react';
import TestRenderer from 'react-test-renderer';

import Cheat from './Cheat';

describe('Cheat component', () => {
  const props = {
    cheat: {
      description: 'cheat description',
      command: 'cheat command',
    }
  };
  it('should render component successfully', () => {
    const renderer = TestRenderer.create(<Cheat {...props} />);

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
