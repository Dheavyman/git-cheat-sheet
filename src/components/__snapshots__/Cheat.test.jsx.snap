// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cheat component should render component successfully 1`] = `
<div
  className="cheat"
>
  <p
    className="cheat-description"
  >
    cheat description
  </p>
  <p
    className="cheat-command"
  >
    $Â 
    cheat command
  </p>
</div>
`;
